src/main/java/com/google/sps/utils/UserUtils.java

```diff
@@ -150,4 +151,18 @@ public static void addPoints(Entity userEntity, int numPoints, DatastoreService
     }
     datastore.put(userEntity);
   }
+
+  /**
+    Adds 20 points to a user for uploading if it has been more than a day since they last uploaded
+  */
+  public static void addUploadPoints(Entity userEntity, DatastoreService datastore){
+
+    QuizTimingPropertiesUtils utils = new QuizTimingPropertiesUtils();
+    if(userEntity.getProperty("lastUploadTime") == null || utils.isTimestampOutdated((long) userEntity.getProperty("lastUploadTime"))){
```

Comment:
should they be getting points if they never uploaded?

Reply:
This function is only called after they upload (if lastUploadTime is null is means it's never gone into that loop before, so that means this is the first time they're uploading)